container-storage-setup.sh

```diff
@@ -731,40 +729,47 @@ create_partition_parted(){
 }
 
 create_partition() {
-  local dev="$1"
+  local dev="$1" part
 
   if [ -x "/usr/sbin/parted" ]; then
     create_partition_parted $dev
   else
     create_partition_sfdisk $dev
   fi
 
+  part=$(query_first_child $dev)
```

Comment:
Okay makes sense. The `BLKRRPART` ioctl is a separate issue, I will drop that from the scope of this PR. @rhvgoyal Any specific reason you introduced `udevadm settle` and `wait_for_dev` ? Since if `create_partition_sfdisk` or `create_partition_parted` is successful, the device for the partition will exist. So we can just call `query_first_child()` and extract the partition name.